### PR TITLE
feat: Implement miles/kilometer selector

### LINKED ISSUE #244: Allow users to change between miles and km
Right now, it's showing `mi` when user's language is `en-US`, and `km` everywhere else.

My phone is en-US, but I use km. I would like to be able to configure that.

This setting should be saved in AsyncStorage.

For the UI, I think for now a simple [Switch](https://facebook.github.io/react-native/docs/switch) in the About.tsx screen would do.

### PR DESCRIPTION
**Pull request by matepapp**
### Prettier
~I tried to infer the project's code formatting rules, but I couldn't figure it out completely, that's why I additionally created a `prettierrc` to configure prettier. If it's too much, please comment on it and I'll remove.~

Ohh I've just found [an issue](https://github.com/prettier/prettier/issues/3845) why you might didn't want to use prettier at the first case. I removed the prettier config and sorry if I messed up your formatting 😶 

### Screen recording 

Image 2

### CODE REVIEW

File: App/App.tsx [hunk 1]
```diff
@@ -26,7 +26,8 @@ import {
   ApiContextProvider,
   ErrorContextProvider,
   FrequencyContextProvider,
-  LocationContextProvider
+  LocationContextProvider,
+  DistanceUnitProvider
```
Comment (amaurymartiny) [inline App/App.tsx]: nit: Could you reorder imports by alphabetical order?

File: App/Screens/Home/Header/Header.tsx [hunk 2]
```diff
@@ -42,33 +33,24 @@ interface HeaderProps {
...
-    'haversine_distance_unit'
-  ) as DistanceUnit;
-  const distance = distanceToStation(
-    currentLocation!,
-    api!,
-    haversineDistanceUnit
-  );
+  const shortDistanceUnit = localizedDistanceUnit('short');
+  const distance = distanceToStation(currentLocation!, api!, distanceUnit);
+  console.log('distance', distance + distanceUnit);
```
Comment (amaurymartiny) [inline App/Screens/Home/Header/Header.tsx]: Remove console.log

File: App/Screens/Home/Header/Header.tsx [hunk 3]
```diff
@@ -42,33 +33,24 @@ interface HeaderProps {
 export function Header (props: HeaderProps) {
   const { api } = useContext(ApiContext)!;
   const { currentLocation, isGps } = useContext(CurrentLocationContext);
+  const { localizedDistanceUnit, distanceUnit } = useDistanceUnit();
```
Comment (amaurymartiny) [inline App/Screens/Home/Header/Header.tsx]: alphabetical

File: App/Screens/About/About.tsx [hunk 4]
```diff
@@ -77,35 +69,26 @@ interface AboutProps
 
 export function About (props: AboutProps) {
   const { navigation } = props;
+  const { distanceUnit, setDistanceUnit, localizedDistanceUnit } = useDistanceUnit();
```
Comment (amaurymartiny) [inline App/Screens/About/About.tsx]: alphabetical

File: App/stores/distanceUnit.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  localizedDistanceUnit: () => '',
+  setDistanceUnit: () => {}
+});
+
+export const DistanceUnitProvider: FC = ({ children }) => {
+  const [distanceUnit, setDistanceUnit] = useState<DistanceUnit>('km');
+
+  const getDistanceUnit = async (): Promise<void> => {
+    const unit = await AsyncStorage.getItem(STORAGE_KEY);
+    if ((unit && unit === 'km') || (unit && unit === 'mile')) {
```
Comment (amaurymartiny) [inline App/stores/distanceUnit.tsx]:
```suggestion
    if (unit === 'km' || unit === 'mile') {
```

File: App/stores/index.ts [hunk 6]
```diff
@@ -18,3 +18,4 @@ export * from './api';
 export * from './error';
 export * from './frequency';
 export * from './location';
+export * from './distanceUnit';
```
Comment (amaurymartiny) [inline App/stores/index.ts]: alphabetical

File: App/Screens/About/About.tsx [hunk 7]
```diff
@@ -33,6 +26,7 @@ import { BackButton } from '../../components';
 import { i18n } from '../../localization';
 import { trackScreen } from '../../util/amplitude';
 import * as theme from '../../util/theme';
+import { useDistanceUnit } from '../../stores/distanceUnit';
```
Comment (amaurymartiny) [inline App/Screens/About/About.tsx]: alphabetical

File: App/Screens/Details/Details.tsx [hunk 8]
```diff
@@ -27,11 +27,8 @@ import { Header } from './Header';
 import { i18n } from '../../localization';
 import { ApiContext, CurrentLocationContext } from '../../stores';
 import { trackScreen } from '../../util/amplitude';
-import {
-  distanceToStation,
-  getCorrectLatLng,
-  DistanceUnit
-} from '../../util/station';
+import { distanceToStation, getCorrectLatLng } from '../../util/station';
+import { useDistanceUnit } from '../../stores/distanceUnit';
```
Comment (amaurymartiny) [inline App/Screens/Details/Details.tsx]: alphabetical

File: App/stores/distanceUnit.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import { AsyncStorage } from 'react-native';
+import React, { useState, useEffect, createContext, FC, useContext } from 'react';
```
Comment (amaurymartiny) [inline App/stores/distanceUnit.tsx]: alphabetical

File: App/stores/distanceUnit.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  localizedDistanceUnit: (format: DistanceUnitFormat) => string;
+};
+
+const Context = createContext<ContextType>({
+  distanceUnit: 'km',
+  localizedDistanceUnit: () => '',
+  setDistanceUnit: () => {}
+});
+
+export const DistanceUnitProvider: FC = ({ children }) => {
```
Comment (amaurymartiny) [inline App/stores/distanceUnit.tsx]: nit: I use the `function` notation everywhere in this project, so would like to see it here too.

File: App/stores/distanceUnit.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+  const [distanceUnit, setDistanceUnit] = useState<DistanceUnit>('km');
```
Comment (amaurymartiny) [inline App/stores/distanceUnit.tsx]: The default distanceUnit should be `mile` if i18n's i18n.locale is `en-US`, or `km` in all other cases.

File: App/Screens/About/About.tsx [hunk 12]
```diff
@@ -169,6 +142,18 @@ export function About (props: AboutProps) {
         </Text>
       </View>
 
+      <View style={styles.distance}>
+        <Text style={styles.h2}>{i18n.t('about_distance_unit_title')}</Text>
+        <View style={styles.distanceSwitchWrapper}>
+          <Switch
+            onValueChange={toggleDistanceSwitch}
+            value={distanceUnit === 'km'}
```
Comment (amaurymartiny) [inline App/Screens/About/About.tsx]: reorder props alphabetically

**Comment by matepapp** [comment]
No problem, thanks for the comments 🙌 I think I fixed everything but double check it please.

**Pull request merged**